### PR TITLE
feat: debounce results for contact search

### LINKED ISSUE #1907: debounce contact search in messages

### PR DESCRIPTION
**Pull request by jim4067**
Resolves #1907

### CODE REVIEW

**Comment by hkirat** [comment]
Thankyou for your PR @jim4067 
Can you take the approach similar to the one taken here - https://github.com/coral-xyz/backpack/blob/master/packages/app-extension/src/components/Unlocked/Messages/ChatDrawer.tsx#L63 (bonus if you can move this functionality to a common folder) and decrease the time to 250ms?

**Comment by jim4067** [comment]
Cool. Let me take care of it.

**Comment by jim4067** [comment]
@hkirat are the recent changes, what you had in mind? 
using your approach it seems that it isn't debouncing. Am I missing something?

File: packages/message-sdk/src/components/Inbox.tsx [hunk 1]
```diff
@@ -33,28 +33,39 @@ export function Inbox() {
       !activeChats.map((x) => x.remoteUsername).includes(result.username)
   );
 
+  let debouncedTimer;
```
Comment (hkirat) [inline packages/message-sdk/src/components/Inbox.tsx:36]: Can you make this variable global, that might be why this isn't working

Comment (jim4067) [reply packages/message-sdk/src/components/Inbox.tsx]: thanks. that was indeed the issues

**Review by hkirat — APPROVED** [review]

**Pull request merged**